check/misc

```diff
@@ -12,7 +12,7 @@ cd "$( dirname "${BASH_SOURCE[0]}" )"
 cd "$(git rev-parse --show-toplevel)"
 
 # Check for non-contrib references to contrib.
-results=$(grep -Rl "\bcirq\.contrib\b" cirq | grep -v "cirq/contrib" | grep -v "__")
+results=$(grep -Rl "\bcirq\.contrib\b" cirq-core | grep -v "cirq/contrib" | grep -v "__")
```

Comment:
Should this be `cirq-core/contrib`?

Reply:
Nope, the path to contrib is `cirq-core/cirq/contrib`.

Reply:
I think we need to change `grep -v "cirq/contrib"` to `grep -v "cirq-core/contrib"`. https://github.com/vtomole/Cirq/commit/e306666f1132fb6f9e9f4cb74bdc5562914f8b61 passes the `misc` check on my machine with this code.

Reply:
I think that's because the commit imports `contrib`, but the misc script is looking for references to `cirq.contrib`. Here's what I get

```
$ git diff
diff --git a/cirq-core/cirq/ops/arithmetic_operation.py b/cirq-core/cirq/ops/arithmetic_operation.py
index d9faf384..2dc3746c 100644
--- a/cirq-core/cirq/ops/arithmetic_operation.py
+++ b/cirq-core/cirq/ops/arithmetic_operation.py
@@ -20,6 +20,7 @@ from typing import Union, Iterable, List, Sequence, cast, TypeVar, TYPE_CHECKING
 import numpy as np
 
 from cirq.ops.raw_types import Operation
+import cirq.contrib
 
 if TYPE_CHECKING:
     import cirq
$ ./check/misc
'cirq.contrib' mentioned in non-contrib files:
cirq-core/cirq/ops/arithmetic_operation.py

Source files outside cirq.contrib must not reference cirq.contrib, as this can cause errors for users who 'pip install cirq' instead of 'pip install cirq-core[contrib]'.
$
```

Reply:
Also, note that `cirq-core/contrib` does not exist, so `grep -v cirq-core/contrib` is unlikely to have any effect on the output. OTOH, `grep -v cirq/contrib` removes references to `cirq.contrib` found under `cirq-core/cirq/contrib` (which is one place where we allow them).